multicluster/controllers/multicluster/leader/clusterset_controller.go

```diff
@@ -72,10 +74,18 @@ func (r *LeaderClusterSetReconciler) Reconcile(ctx context.Context, req ctrl.Req
 			return ctrl.Result{}, err
 		}
 		klog.InfoS("Received ClusterSet delete", "clusterset", req.NamespacedName)
+		if r.clusterSetConfig != nil && r.clusterSetConfig.Name != req.Name {
+			return ctrl.Result{}, nil
+		}
 		r.clusterSetConfig = nil
 		r.clusterID = common.InvalidClusterID
 		r.clusterSetID = common.InvalidClusterSetID
-
+		if err := cleanUpAllMemberClusterAnnounces(ctx, r.Client); err != nil {
```

Comment:
You are right, we have a webhook to prevent ClusterSet deletion in the leader cluster if there is a memberClusterAnnounce, this is unnecessary. Removed it.